lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1244,6 +1243,11 @@ dateLoop:
 			}
 			values = append(values, result...)
 			if limitReached {
+				// If we achieved the limit, we need to check if we have more events to fetch from the current date
+				// or if we need to move the cursor to the next date.
+				if hasLeft() && len(l.checkpoint.Iterator) == 0 {
```

Comment:
nit: I'd just expand the hasLeft here just to make it clear that we're literally just checking if we can index into `l.dates[i+1]`.